### ISSUE TITLE
Git opening an error window after incorrect entry

### ISSUE THREAD

**Issue by lrmendes** [git-for-windows/git#3308]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?
 
```
64-bit version.

$ git --version --build-options

git version 2.28.0.windows.1
cpu: x86_64
built from commit: 77982caf269b7ee713a76da2bcf260c34d3bf7a7
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh

```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.19042.1052]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: VIM
Custom Editor Path: 
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Merge
Use Credential Manager: Enabled
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?
```
No 
```

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

```
CMD (Command Prompt)
```

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
1. Open the Windows CMD (Command Prompt).
2. Run the following command: git --version --help
3. A window message box will be displayed with the message: "The system cannot find the file C:/Program Files/Git/mingw64/share/doc/git-doc/git-version.html."
```
![bug-visualization](https://user-images.githubusercontent.com/14244140/124369118-625a7a80-dc60-11eb-9728-b91620b0039a.png)

 - What did you expect to occur after running these commands?
```
Error messages for incorrect entries should be displayed in the CLI (Terminal).
```

 - What actually happened instead?

```
Git opens an error window with the message: "The system cannot find the file C:/Program Files/Git/mingw64/share/doc/git-doc/git-version.html."
```

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

```
N/A
```

**Comment by PhilipOakley** [comment]
> . Open the Windows CMD (Command Prompt).
> 2. Run the following command: git --version --help
> 3. A window message box will be displayed with the message: "The system cannot find the file C:/Program Files/Git/mingw64/share/doc/git-doc/git-version.html."

This is (I believe) 'acting as designed', but maybe it could be better.

Basically the `--help` option wins and tries to open the implied command's manual page, which on Git for Windows is via that file, but given the command line is only options it gets it wrong.

You have a number of options depending on what you hope for,  but mainly:
`git help git`  deliberately run the help command and ask about `git`, and see that it has the `--version` option. It (git) will also respond to pseudonymous `version` command is if the --version was given, even if it doesn't have a man page.

One option (as a contributor) is to create a templated `git version` man page, just so that this issue disappears and we capture, for the reader, that info. There may be some Catch-22 issues, but we can already see that Git does try to look for that man page.

**Comment by rimrul** [comment]
> acting as designed

yes, but the design is subpar. The `--help``option just like `git help` tries to determine the help page to show and either calls `man git-foo` or ask the operating system to open `/mingw64/share/doc/git-doc/git-foo.html`. The OS then complains about the file not existing in a message box. For `man` based help a non-existing page isn't a UX issue, `man` will just complain that the page doesn't exist. For HTML based help Windows will show a MessageBox for non-existing help pages. We could probably check if the file we want to open actually exists and show an appropriate error message if it doesn't. 

[https://github.com/git/git/blob/142430338477d9d1bb25be66267225fb58498d92/builtin/help.c#L472-L477](We already make sure the directory exists, but don't check the actual file.)

**Comment by dscho** [comment]
My biggest question, though, is: is this a Windows-specific issue? If not, it would make most sense to take it to [the Git mailing list](mailto:git@vger.kernel.org) (send plain-text messages, HTML messages are dropped silently).

**Comment by rimrul** [comment]
Yes and no. This specific error message is windows specific (since it's generated by windows). The absence of a `git version` help page and the fact that we don't check wether the help file exists before passing it to one of the "browser candidates", in this case `start` is not windows specific.

**Comment by dscho** [comment]
> This specific error message is windows specific (since it's generated by windows).

I don't actually think so. `get_html_page_path()` is compiled also on non-Windows platforms, and while the default for `git help` is to show manual pages there, users _can_ configure to show HTML pages in their web browser instead.

**Comment by rimrul** [comment]
> > This specific error message is windows specific (since it's generated by windows).

> I don't actually think so. `get_html_page_path()` is compiled also on non-Windows platforms

The messagebox and it's contents are generated by `start`. `open`, `cygstart`, `lynx` etc. probably throw a different message.

**Comment by rimrul** [comment]
I'll draft a PR to address both issues with upstream.

**Comment by rimrul** [comment]
This has been fixed in Git 2.34.0
